src/openvic-simulation/economy/production/ResourceGatheringOperation.cpp

```diff
@@ -123,6 +100,51 @@ fixed_point_t ResourceGatheringOperation::calculate_size_modifier(ProvinceInstan
 	return size_modifier > fixed_point_t::_0() ? size_modifier : fixed_point_t::_0();
 }
 
+void ResourceGatheringOperation::rgo_tick(ProvinceInstance& location, ModifierEffectCache const& modifier_effect_cache) {
+	if (production_type_nullable == nullptr) {
+		output_quantity_yesterday = 0;
+		revenue_yesterday = 0;
+		return;
+	}
+	
+	ProductionType const& production_type = *production_type_nullable;
+	std::vector<Job> const& jobs = production_type.get_jobs();
+	IndexedMap<PopType, Pop::pop_size_t> const& province_pop_type_distribution = location.get_pop_type_distribution();
+	
+	Pop::pop_size_t total_worker_count_in_province = 0; //not counting equivalents
+	for(Job const& job : jobs) {
+		total_worker_count_in_province += province_pop_type_distribution[*job.get_pop_type()];
+	}
+	
+	hire(location, total_worker_count_in_province);
+
+	Pop::pop_size_t total_owner_count_in_state_cache = 0;
+	std::vector<Pop*>* owner_pops_cache = nullptr;
```

Comment:
Unless we're planning on adding/removing pops from this vector, we can use a `const*`. We can still access mutable `Pop*`s via a `const*` vector.